shell/platform/darwin/macos/framework/Source/FlutterAccessibilityMac.mm

```diff
@@ -0,0 +1,575 @@
+
+  NSString* mac_notification = nil;
+
+  switch (event_type) {
+    case ax::AXEventGenerator::Event::ACTIVE_DESCENDANT_CHANGED:
+      if (GetData().role == ax::Role::kTree) {
+        mac_notification = NSAccessibilitySelectedRowsChangedNotification;
+      } else if (GetData().role == ax::Role::kTextFieldWithComboBox) {
+        // Even though the selected item in the combo box has changed, we don't
+        // want to post a focus change because this will take the focus out of
+        // the combo box where the user might be typing.
+        mac_notification = NSAccessibilitySelectedChildrenChangedNotification;
+      } else {
+        // In all other cases we should post
+        // |NSAccessibilityFocusedUIElementChangedNotification|, but this is
+        // handled elsewhere.
+        return;
+      }
+      break;
+    case ax::AXEventGenerator::Event::LOAD_COMPLETE:
+      // On MacOS 10.15, firing AXLoadComplete causes focus to move to the
+      // webpage and read content, despite the "Automatically speak the webpage"
+      // checkbox in Voiceover utility being unchecked. The checkbox is
+      // unchecked by default in 10.15 so we don't fire AXLoadComplete events to
+      // support the default behavior.
+      if (SystemVersionEqualTo({.majorVersion = 10, .minorVersion = 15}))
+        return;
+      mac_notification = NSAccessibilityLoadCompleteNotification;
+      break;
+    case ax::AXEventGenerator::Event::INVALID_STATUS_CHANGED:
+      mac_notification = NSAccessibilityInvalidStatusChangedNotification;
+      break;
+    case ax::AXEventGenerator::Event::SELECTED_CHILDREN_CHANGED:
+      if (ax::IsTableLike(GetData().role)) {
+        mac_notification = NSAccessibilitySelectedRowsChangedNotification;
+      } else {
+        // VoiceOver does not read anything if selection changes on the
+        // currently focused object, and the focus did not move. Fire a
+        // selection change if the focus did not change.
+        NSAccessibilityElement* native_accessibility_node = (NSAccessibilityElement*)native_node;
+        if (native_accessibility_node.accessibilityFocusedUIElement &&
+            GetData().HasState(ax::State::kMultiselectable) &&
+            !IsInGeneratedEventBatch(ax::AXEventGenerator::Event::ACTIVE_DESCENDANT_CHANGED) &&
+            !IsInGeneratedEventBatch(ax::AXEventGenerator::Event::FOCUS_CHANGED)) {
+          // Force announcement of current focus / activedescendant, even though
+          // it's not changing. This way, the user can hear the new selection
+          // state of the current object. Because VoiceOver ignores focus events
+          // to an already focused object, this is done by destroying the native
+          // object and creating a new one that receives focus.
+
+          // static_cast<BrowserAccessibilityMac*>(focus)->ReplaceNativeObject();
+
+          // Don't fire selected children change, it will sometimes override
+          // announcement of current focus.
```

Comment:
We should either remove this commented out code or uncomment it.

If this is functionality we just don't care about, remove it.

If we're not sure if we care about it, add a TODO and file a bug to track that down.

If we know we need to implement this but don't know how to do right now, add  a TODO and file a bug.